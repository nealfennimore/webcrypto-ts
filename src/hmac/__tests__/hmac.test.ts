import { HMAC, HmacKey } from "../index";

describe("HMAC", () => {
    let key: HmacKey;
    beforeEach(async () => {
        key = await HMAC.generateKey();
    });
    it("should generate key", async () => {
        expect(key).toMatchSnapshot();
    });
    it("should import and export key", async () => {
        let jwk = await HMAC.exportKey("jwk", key);
        const importedPubKey = await HMAC.importKey("jwk", jwk, {
            hash: "SHA-512",
        });

        expect(await HMAC.exportKey("jwk", importedPubKey)).toEqual(jwk);
    });
    it("should sign and verify", async () => {
        const text = encode("a message");
        const signature = await HMAC.sign(key, text);

        expect(await HMAC.verify(key, signature, text)).toBe(true);
    });
});
