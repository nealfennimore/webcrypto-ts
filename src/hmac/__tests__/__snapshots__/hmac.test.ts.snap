// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HMAC should generate key 1`] = `
CryptoKey {
  Symbol(kKeyObject): SecretKeyObject {
    Symbol(kKeyType): "secret",
  },
  Symbol(kAlgorithm): Object {
    "hash": Object {
      "name": "SHA-512",
    },
    "length": 1024,
    "name": "HMAC",
  },
  Symbol(kExtractable): true,
  Symbol(kKeyUsages): Array [
    "sign",
    "verify",
  ],
}
`;
